### PR TITLE
Handle stop codons and update benchmarks

### PR DESCRIPTION
**Pull request by jgarciamesa**
Early stop codons are not allowed in the ancestor sequence. Ending stop codons are handled post-alignment.
Modify the substitution matrix for Muse and Gaut. The instantaneous substitution rate (Q matrix values) from stop codons is zero.
Change the names of models to tri-mg, tri-ecm, mar-mg, mar-ecm, and dna.
Simplify alignment output by converting alignment FST to strings beforehand.

### CODE REVIEW

**Comment by jgarciamesa** [comment]
This PR now also includes fixes:
 - Preserve sequence order when outputting to json (previously ordered alphabetically by default).
 - Correctly display output to console.
 - Check all codons (except last) for early stop codons in ancestor.

and updates benchmark tests by:
 - Removing duplicates.
 - Removing early stop codon in 32k file.
 - Running 10 iterations per test.

File: src/lib/utils.cc [hunk 1]
```diff
@@ -682,4 +724,247 @@ TEST_CASE("get_nuc") {
...
+/**
+ * @brief Trim end stop codons.
+ *
+ * @param[in,out] aln coati::alignment_t input sequences and alignment info.
+ */
+void trim_end_stops(coati::data_t& data) {
+    for(size_t i = 0; i < data.size(); ++i) {
+        std::string seq{data.seqs[i]};
+        size_t len = seq.length();
+        std::string last_cod{seq.substr(len - 3)};
```
Comment (reedacartwright) [inline src/lib/utils.cc:793]: You should be able to use a string_view here.

File: src/lib/utils.cc [hunk 2]
```diff
@@ -682,4 +724,247 @@ TEST_CASE("get_nuc") {
...
+ * @brief Restore end stop codons post pairwise alignment.
+ *
+ * @details Four case scenarios:
+ *  (1) no end stop codons on either sequence: nothing to do.
+ *  (2) only present in one sequence: add codon back and insert 3 gaps to other
+        sequence.
+ *  (3) present in both and same stop codon: add codons back as 3 matches.
+ *  (4) present in both but different stop codon: add codons as a
+ *      3 len insertion + a 3 len deletion.
+ *
```
Comment (reedacartwright) [inline src/lib/utils.cc:875]: For case (4), I think it would be most parsimonious to insert them back as 3 matches. Stop codons do evolve.

**Review by reedacartwright — COMMENTED** [review]
Here's a quick review.

File: src/lib/utils.cc (hunk 1 above)
Comment (jgarciamesa) [reply src/lib/utils.cc]: Done, thanks.

File: src/lib/utils.cc (hunk 2 above)
Comment (jgarciamesa) [reply src/lib/utils.cc:873]: Makes sense, I agree.

**Pull request merged**